community/kernel-api/src/test/java/org/neo4j/internal/kernel/api/NodeValueIndexCursorTestBase.java

```diff
@@ -480,6 +481,86 @@ public void shouldGetNoIndex() throws Exception
+        try ( org.neo4j.internal.kernel.api.Transaction tx = session.beginTransaction();
+              NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor() )
+        {
+            // when
+            tx.dataWrite().nodeDelete( strOne );
+            tx.dataRead().nodeIndexSeek( index, node, IndexOrder.NONE, IndexQuery.exact( prop, "one" ) );
+
+            // then
+            assertFalse( node.next() );
+        }
+    }
+
+    @Test
+    public void shouldNotFindDeletedNodeInIndexScan() throws Exception
+    {
+        // Given
+        int label = token.nodeLabel( "Node" );
+        int prop = token.propertyKey( "prop" );
+        CapableIndexReference index = schemaRead.index( label, prop );
+        IndexValueCapability wildcardCapability = index.valueCapability( ValueGroup.UNKNOWN );
+        try ( org.neo4j.internal.kernel.api.Transaction tx = session.beginTransaction();
+              NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor();
+              PrimitiveLongSet uniqueIds = Primitive.longSet() )
+        {
+            // when
+            tx.dataRead().nodeIndexScan( index, node, IndexOrder.NONE );
+            assertFoundNodesAndValue( node, 24, uniqueIds, wildcardCapability );
+
+            // then
+            tx.dataWrite().nodeDelete( strOne );
+            tx.dataRead().nodeIndexScan( index, node, IndexOrder.NONE );
+            assertFoundNodesAndValue( node, 23, uniqueIds, wildcardCapability );
+        }
+    }
+
+    @Test
+    public void shouldNotFindUpdatedNodeInIndexSeek() throws Exception
+    {
+        // Given
+        int label = token.nodeLabel( "Node" );
+        int prop = token.propertyKey( "prop" );
+        CapableIndexReference index = schemaRead.index( label, prop );
+        try ( org.neo4j.internal.kernel.api.Transaction tx = session.beginTransaction();
+              NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor() )
+        {
+            // when
+            tx.dataWrite().nodeSetProperty( strOne, prop, stringValue("ett") );
+            tx.dataRead().nodeIndexSeek( index, node, IndexOrder.NONE, IndexQuery.exact( prop, "one" ) );
+
+            // then
+            assertFalse( node.next() );
+        }
+    }
+
+    @Test
+    public void shouldFindUpdatedNodeInIndexSeek() throws Exception
+    {
+        // Given
+        int label = token.nodeLabel( "Node" );
+        int prop = token.propertyKey( "prop" );
+        CapableIndexReference index = schemaRead.index( label, prop );
+        try ( org.neo4j.internal.kernel.api.Transaction tx = session.beginTransaction();
+              NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor() )
+        {
+            // when
+            tx.dataWrite().nodeSetProperty( strOne, prop, stringValue("ett") );
+            tx.dataRead().nodeIndexSeek( index, node, IndexOrder.NONE, IndexQuery.exact( prop, "ett" ) );
+
+            // then
+            assertTrue( node.next() );
+        }
+    }
```

Comment:
I don't remember what other tx-state tests we have for index reading. So I'm going to list some other things I think we might be interested in testing if they are not tested in other places:

 * Deleted nodes and range/prefix scans
 * Some composite index queries
 * Added/removed label with various query types
 * Swapping a indexed node in a tx (remove prop from (a:L), add to (b:L))